lib/yunohost/tools.py

```diff
@@ -469,3 +474,44 @@ def tools_upgrade(auth, ignore_apps=False, ignore_packages=False):
     if is_api:
         from yunohost.service import service_log
         return { "log": service_log('yunohost-api', number="100").values()[0] }
+
+
+def tools_upgrade_v24(auth):
+    """
+    YunoHost upgrade to new Yunohost version (on jessie)
+    """
+    import platform
+
+    # Retrieve interface
+    is_api = True if msettings.get('interface') == 'api' else False
+
+    # Get Debian major version
+    debian_major_version = platform.linux_distribution()[1].split('.')[0]
+    if debian_major_version is not '8':
+        msignals.display(m18n.n('upgrade_24_not_jessie'), 'error')
+        return
+
+    # Upgrade with current sources
+    os.system('apt-get update')
+    os.system('yes "q" | DEBIAN_FRONTEND=noninteractive DEBIAN_PRIORITY=critical apt-get -y --force-yes -qq -o Dpkg::Options::="--force-confdef" -o Dpkg::Options::="--force-confold" upgrade')
```

Comment:
os.system won't raised an exception if the program fails (add to patch that for prosopopee today).

A way to ensure you don't continued in a broken state woudl be to do a 

```python
assert os.system(command) == 0, "The command '%s' failed, abort" % command
```

To check that the return code of the command is 0.